taichi/backends/struct_llvm.cpp

```diff
@@ -263,6 +269,10 @@ void StructCompilerLLVM::run(SNode &root, bool host) {
     auto set_assert_failed = tlctx->lookup_function<void(void *, void *)>(
         "Runtime_set_assert_failed");
 
+    auto set_vvprintf =
```

Comment:
Is there a reason to use `vvprintf` instead of `vprintf`?

Reply:
To avoid possible name conflicts in some env, just in case. Some headers like to define names as macros, eg. my `/usr/include/stdio.h`:
```c
#if !__GLIBC_USE (DEPRECATED_SCANF) && !defined __LDBL_COMPAT
# ifdef __REDIRECT
extern int __REDIRECT (fscanf, (FILE *__restrict __stream,
                                const char *__restrict __format, ...),
                       __isoc99_fscanf) __wur;
extern int __REDIRECT (scanf, (const char *__restrict __format, ...),
                       __isoc99_scanf) __wur;
extern int __REDIRECT_NTH (sscanf, (const char *__restrict __s,
                                    const char *__restrict __format, ...),
                           __isoc99_sscanf);
# else
extern int __isoc99_fscanf (FILE *__restrict __stream,
                            const char *__restrict __format, ...) __wur;
extern int __isoc99_scanf (const char *__restrict __format, ...) __wur;
extern int __isoc99_sscanf (const char *__restrict __s,
                            const char *__restrict __format, ...) __THROW;
#  define fscanf __isoc99_fscanf
#  define scanf __isoc99_scanf
#  define sscanf __isoc99_sscanf
# endif
#endif
```

What's more:
```c
/* Standard streams.  */
extern FILE *stdin;             /* Standard input stream.  */
extern FILE *stdout;            /* Standard output stream.  */
extern FILE *stderr;            /* Standard error output stream.  */
/* C89/C99 say they're macros.  Make them happy.  */
#define stdin stdin
#define stdout stdout
#define stderr stderr
```
Make them happy(?)

Anyway, runtime compiled code should pays more attention to compatibility, since not only we developers use them, but also pip users.

Reply:
It's sad that legacy code can hardly make everybody happy. They make C99 happy, but now we have to do extra work... :-)

Reply:
I understand that `vvprintf` probably leads to less conflicts compared to `vprintf`, but simply duplicating `v` here is a bit random. How about `vprintf_host`? This indicates that we are calling a host function. Note that on CUDA we still want to use the built-in `vprintf` function.